addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/providers/KubernetesClientProvider.java

```diff
@@ -0,0 +1,31 @@
+package io.fabric8.launcher.osio.providers;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Produces;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.HttpHeaders;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.osio.EnvironmentVariables;
+
+import static io.fabric8.launcher.base.identity.IdentityFactory.createFromToken;
+import static io.fabric8.launcher.base.identity.TokenIdentity.removeBearerPrefix;
+
+@RequestScoped
+public class KubernetesClientProvider {
+
+    @Produces
+    public KubernetesClient createOpenshiftClient(HttpServletRequest request) {
```

Comment:
Not needed. You can use @Inject OpenshiftService and call getOpenshiftClient

Reply:
fixed in #84